dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/TrnLookupPageModel.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeacherIdentity.AuthServer.Services.DqtApi;
+
+namespace TeacherIdentity.AuthServer.Pages.SignIn.Trn;
+
+public abstract class TrnLookupPageModel : PageModel
+{
+    private static readonly TimeSpan _trnLookupTimeout = TimeSpan.FromSeconds(5);
+
+    private readonly ILogger<TrnLookupPageModel> _logger;
+
+    protected TrnLookupPageModel(
+        IIdentityLinkGenerator linkGenerator,
+        IDqtApiClient dqtApiClient,
+        ILogger<TrnLookupPageModel> logger)
+    {
+        LinkGenerator = linkGenerator;
+        DqtApiClient = dqtApiClient;
+        _logger = logger;
+    }
+
+    public IIdentityLinkGenerator LinkGenerator { get; }
+
+    public IDqtApiClient DqtApiClient { get; }
+
+    protected async Task<IActionResult?> TryFindTrn()
```

Comment:
Are we calling this every time we POST in the new TRN flow? Can we not check whether the lookup has previously been completed?